app/components/recognition/form.js

```diff
@@ -311,4 +320,114 @@ export default class FormComponent extends Component {
     }
     return awardedBy;
   }
+  @action
+  async handleFileChange(event) {
+    this.legalResourceFile = event.target.files[0];
+    if (this.legalResourceFile) {
+      this.currentRecognition.recognitionModel.file = this.legalResourceFile;
+      this.currentRecognition.recognitionModel.file.download =
+        this.legalResourceFile;
+      set(this.validationErrors, 'legalResource', null);
+      this.currentRecognition.recognitionModel.file.isNew = true;
+    }
+  }
+
+  removeFile = task({ drop: true }, async () => {
+    try {
+      let file = await this.legalResourceFile;
+      if (!file.id) {
+        this.legalResourceFile = null;
+        this.clearFormError('legalResource');
+        this.clearFormError('file');
+        await this.currentRecognition.recognition.setProperties({
+          ...this.currentRecognition.recognition,
+          legalResource: null,
+          file: null,
+        });
+        return set(this.validationErrors, 'legalResource', null);
+      }
+      await file.deleteRecord();
```

Comment:
I'm not sure if we want to do this here? At the moment the file relationship will actually be removed, even if the user doesn't save the form.

I would have expected this to only happen when "save" is pressed by the user.

Reply:
@Windvis thanks for your review. I changed it to only delete the file on save.